Reintroduce PR #5974: Remove klogr dependency and move to zap

aliaqel-stripe opened the pull request:
<!-- Thank you for contributing!

     Read more about how you can contribute in our contribution guide:
     https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md
-->

_Provide a description of what has been changed_

### Checklist

- [ ] When introducing a new scaler, I agree with the [scaling governance policy](https://github.com/kedacore/governance/blob/main/SCALERS.md)
- [ ] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [ ] Tests have been added
- [ ] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [ ] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [ ] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [ ] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

<!--
  Make sure to link the related issue for this change
  If it requires multiple PRs and/or a PR on another repo as well, please use "Relates to" instead.
-->
Fixes #

<!--
  Make sure to link the related PRs for changes such as documentation & Helm charts
-->
Relates to #

aliaqel-stripe commented:
this change appears to have introduced a memory leak, will diagnose and then reach out to review

lucasfcnunes commented:
> In general, this is great.
> 
> My concern is backwards compatibilty, we are changing the way to setup this. I wonder if there's a way how to bypass this, maybe internally convert the "old" params to the new one?

Yep. That's how it needs to be.

aliaqel-stripe commented:
@zroubalik @lucasfcnunes I added handling for the legacy zap flags KEDA exposes through Helm and converted them to zap options to the extent that it made sense. Tested them locally on our setup without modifying flags (i.e use the existing klogr flags) and verified logging works and the metrics-server starts up.

zroubalik commented:
/run-e2e internal

aliaqel-stripe commented:
@zroubalik We should probably rerun the e2e tests if possible. Had to rebase to because of a conflict

JorTurFer commented:
/run-e2e internal

aliaqel-stripe commented:
@JorTurFer Not sure why the latest e2e test run failed: https://github.com/kedacore/keda/actions/runs/14682302792

Looks like it was green but the result was failure?

zroubalik commented:
/run-e2e internal

JorTurFer commented:
/run-e2e internal

JorTurFer commented:
/run-e2e internal
